src/io.jl

```diff
@@ -223,9 +226,530 @@ function readg(gfile; set_time=nothing)
+            add_derived=add_derived,
+        )
+    end
+end
+
+"""
+    function geqdsk2imas!(
+        g::GEQDSKFile,
+        dd::IMASdd.dd,
+        time_index::Int;
+        geqdsk_cocos::Int=0,
+        dd_cocos::Int=11,
+        cocos_clockwise_phi::Bool=false,
+        add_derived::Bool=false,
+    )
+
+Utility for writing equilibrium data from a GEQDSKFile to IMAS DD.
+
+$geqdsk2imas_generic
+
+    g: GEQDSKFile instance
+    dd: IMAS data dictionary
+    time_index: index of the destination for equilibrium data within equilibrium.time_slice
+    geqdsk_cocos: coordinate convention indentifier of the incoming GEQDSK set. 0 to auto detect.
+    dd_cocos: coordinate convention identifier of the DD. Should normally be left at 11.
+    cocos_clockwise_phi: hint for deciding COCOS, because GEQDSK probably doesn't constrain this.
+    add_derived: switch to use functions within EFIT.jl to extend equilibrium data
+"""
+function geqdsk2imas!(
+    g::GEQDSKFile,
+    dd::IMASdd.dd,
+    time_index::Int;
+    geqdsk_cocos::Int=0,
+    dd_cocos::Int=11,
+    cocos_clockwise_phi::Bool=false,
+    add_derived::Bool=false,
+)
+    if ismissing(dd.dataset_description.data_entry, :pulse)
+        dd.dataset_description.data_entry.pulse = parse(Int, split(split(g.file, ".")[end-1], "g")[2])
+    end
+    geqdsk2imas!(
+        g, dd.equilibrium, time_index;
+        wall=dd.wall, geqdsk_cocos=geqdsk_cocos, dd_cocos=dd_cocos, cocos_clockwise_phi=cocos_clockwise_phi,
+        add_derived=add_derived,
+    )
+end
+
+"""
+    function geqdsk2imas!(
+        g::GEQDSKFile,
+        eq::IMASdd.equilibrium,
+        time_index::Int;
+        wall=nothing,
+        geqdsk_cocos::Int=0,
+        dd_cocos::Int=11,
+        cocos_clockwise_phi::Bool=false,
+        add_derived::Bool=false,
+    )
+
+Utility for writing equilibrium data from a GEQDSKFile to IMAS equilibrium IDS
+
+$geqdsk2imas_generic
+
+    g: GEQDSKFile instance
+    eq: Reference to equilibrium IDS
+    time_index: index of the destination for equilibrium data within equilibrium.time_slice
+    wall: Optional reference to wall IDS
+    geqdsk_cocos: coordinate convention indentifier of the incoming GEQDSK set. 0 to auto detect.
+    dd_cocos: coordinate convention identifier of the DD. Should normally be left at 11.
+    cocos_clockwise_phi: hint for deciding COCOS, because GEQDSK probably doesn't constrain this.
+    add_derived: switch to use functions within EFIT.jl to extend equilibrium data
+"""
+function geqdsk2imas!(
+    g::GEQDSKFile,
+    eq::IMASdd.equilibrium,
+    time_index::Int;
+    wall=nothing,
+    geqdsk_cocos::Int=0,
+    dd_cocos::Int=11,
+    cocos_clockwise_phi::Bool=false,
+    add_derived::Bool=false,
+)
+    if geqdsk_cocos == 0
+        geqdsk_cocos = CoordinateConventions.identify_cocos(
+            sign(g.bcentr), sign(g.current), sign(g.qpsi[1]), sign(g.psi[end] - g.psi[1]), cocos_clockwise_phi,
+        )[1]
+        println("  identified COCOS=$geqdsk_cocos")
```

Comment:
Please don't pollute peoples' stdout
```suggestion
        @debug "  identified COCOS=$geqdsk_cocos"
```